webapp/src/main/java/com/box/l10n/mojito/android/strings/AndroidStringDocumentMapper.java

```diff
@@ -0,0 +1,186 @@
+package com.box.l10n.mojito.android.strings;
+
+import com.box.l10n.mojito.service.tm.PluralNameParser;
+import com.box.l10n.mojito.service.tm.search.TextUnitDTO;
+import com.google.common.base.Strings;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.box.l10n.mojito.android.strings.AndroidPluralQuantity.OTHER;
+
+public class AndroidStringDocumentMapper {
+
+    private static final String DEFAULT_ASSET_DELIMITER = "#@#";
+
+    private final String pluralSeparator;
+    private final String assetDelimiter;
+    private final String locale;
+    private final String repositoryName;
+
+    public AndroidStringDocumentMapper(String pluralSeparator,
+                                       String assetDelimiter,
+                                       String locale,
+                                       String repositoryName) {
+        this.pluralSeparator = pluralSeparator;
+        this.assetDelimiter = Optional.ofNullable(Strings.emptyToNull(assetDelimiter)).orElse(DEFAULT_ASSET_DELIMITER);
+        this.locale = locale;
+        this.repositoryName = repositoryName;
+    }
+
+    public AndroidStringDocumentMapper(String pluralSeparator, String assetDelimiter) {
+        this(pluralSeparator, assetDelimiter, null, null);
+    }
+
+    public AndroidStringDocument readFromTextUnits(List<TextUnitDTO> textUnits, boolean useSource) {
+
+        AndroidStringDocument document = new AndroidStringDocument();
+        Map<String, AndroidPlural.AndroidPluralBuilder> pluralByOther = new HashMap<>();
```

Comment:
this looks like a mix of "stream" and imperative style. can it be cleaned up?

Reply:
Hmm, this isn't stream style since its using the underneath iterable's `forEach` method. But besides that, implementing this as a Stream isn't gonna work since here we need to maintain a "cache" of `pluralFormOther` to `AndroidPluralBuilder`.

Reply:
it could probably be converted to actual stream but if keeping like that a standard "for loop" makes more sense than the functional style.

Reply:
Updated.